http-server/http-server.go

```diff
@@ -3,34 +3,72 @@ package httpserver
 import (
 	"bytes"
 	"context"
+	"crypto/subtle"
+	"crypto/x509"
 	"encoding/json"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 	"net/http"
+	"os"
 	"runtime"
 	"time"
 
+	grpcRuntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
 	"github.com/packethost/pkg/log"
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/tinkerbell/tink/protos/hardware"
+	"github.com/tinkerbell/tink/protos/template"
+	"github.com/tinkerbell/tink/protos/workflow"
 )
 
 var (
 	gitRev         = "unknown"
 	gitRevJSON     []byte
+	grpcListenAddr = "localhost:42113"
 	httpListenAddr = ":42114"
+	authUsername   = os.Getenv("TINK_AUTH_USERNAME")
+	authPassword   = os.Getenv("TINK_AUTH_PASSWORD")
 	startTime      = time.Now()
 	logger         log.Logger
 )
 
 // SetupHTTP setup and return an HTTP server
 func SetupHTTP(ctx context.Context, lg log.Logger, certPEM []byte, modTime time.Time, errCh chan<- error) {
 	logger = lg
+
+	cp := x509.NewCertPool()
+	ok := cp.AppendCertsFromPEM(certPEM)
+	if !ok {
+		logger.Error(errors.New("parse cert"))
+	}
+
+	creds := credentials.NewClientTLSFromCert(cp, "")
+
+	mux := grpcRuntime.NewServeMux()
+
+	dialOpts := []grpc.DialOption{grpc.WithTransportCredentials(creds)}
+	err := hardware.RegisterHardwareServiceHandlerFromEndpoint(ctx, mux, grpcListenAddr, dialOpts)
+	if err != nil {
+		logger.Error(err)
+	}
+	err = template.RegisterTemplateHandlerFromEndpoint(ctx, mux, grpcListenAddr, dialOpts)
+	if err != nil {
+		logger.Error(err)
+	}
+	err = workflow.RegisterWorkflowSvcHandlerFromEndpoint(ctx, mux, grpcListenAddr, dialOpts)
+	if err != nil {
+		logger.Error(err)
+	}
+
 	http.HandleFunc("/cert", func(w http.ResponseWriter, r *http.Request) {
 		http.ServeContent(w, r, "server.pem", modTime, bytes.NewReader(certPEM))
 	})
 	http.Handle("/metrics", promhttp.Handler())
 	setupGitRevJSON()
 	http.HandleFunc("/version", versionHandler)
 	http.HandleFunc("/_packet/healthcheck", healthCheckHandler)
+	http.Handle("/", BasicAuth(mux))
```

Comment:
I think this should be wrapped in an if statement that checks if `TINK_AUTH_USERNAME` or `TINK_AUTH_PASSWORD` are set.